test/e2e/node/pods.go

```diff
@@ -473,6 +474,62 @@ var _ = SIGDescribe("Pods Extended", func() {
 			}
 			framework.Logf("Summary of latencies:\n%s", buf.String())
 		})
+
+		ginkgo.It("should be consistent between pod spec and status for container image", func() {
+			ginkgo.By("creating the pod")
+			name := "pod-inplace-update-" + string(uuid.NewUUID())
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+					Labels: map[string]string{
+						"name": name,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "busybox",
+							Image: imageutils.GetE2EImage(imageutils.Nginx),
+							Resources: v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceCPU:    resource.MustParse("5m"),
+									v1.ResourceMemory: resource.MustParse("10Mi"),
+								},
+							},
+						},
+					},
+				},
+			}
+
+			ginkgo.By("submitting the pod to kubernetes")
+			podClient.Create(pod)
+
+			ginkgo.By("waiting for pod running")
+			err := e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, pod.Name, f.Namespace.Name)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("updating container image")
+			podClient.Update(name, func(pod *v1.Pod) {
+				pod.Spec.Containers[0].Image = imageutils.GetE2EImage(imageutils.NginxCopy)
```

Comment:
Does this `NginxCopy` image have the same hash as the original `Nginx` image to reproduce the issue?

Reply:
``NginxCopy`` (``nginx-1.14.2-copy``) has the same SHA as the ``Nginx`` image (``nginx-1.14.2``), according to the SHAs promoted: https://github.com/kubernetes/k8s.io/blob/c011745da40d143dfbec89737901143cfed154e7/k8s.gcr.io/images/k8s-staging-e2e-test-images/images.yaml#L78